src/core/preview/preview.ts

```diff
@@ -11,6 +11,11 @@ export class UIPPreview extends UIPPlugin {
   @bind
   protected _onRootStateChange(): void {
     if (this.$inner.parentElement === this) this.removeChild(this.$inner);
+
+    this.model?.html ||
+    this.model?.snippets.length &&
+    this.model?.applySnippet(this.model?.snippets[0], this);
```

Comment:
This line looks complicated.
Short conditions are good but not in the case of the complicated conditions.
Please also be careful with "Elvis", while it's not a native browser feature each usage of the "Elvis" adds an additional condition. Here in the case, the optimizer fails we will have about 4th useless additional checks in the output.

```suggestion
    if (this.model && (this.model.html || this.model.snippets.length)) this.model.applySnippet(this.model.snippets[0], this);
```

And basically, this condition doesn't look like an aria of responsibility for the preview module
It should be handled by the root component or by the model on request.